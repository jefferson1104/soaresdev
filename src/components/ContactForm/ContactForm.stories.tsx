import { Story, Meta } from '@storybook/react/types-6-0'
import ContactForm from '.'

export default {
  title: 'ContactForm',
  component: ContactForm
} as Meta

export const Basic: Story = (args) => <ContactForm {...args} />
