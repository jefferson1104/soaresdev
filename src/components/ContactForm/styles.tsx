import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin: ${theme.spacings.huge} 0;

    > h2 {
      font-size: ${theme.font.sizes.huge};
      color: ${theme.colors.white};
      margin-bottom: ${theme.spacings.xxlarge};
    }

    ${media.lessThan('medium')`
      > h2 {
        font-size: ${theme.font.sizes.xlarge};
      }
    `}
  `}
`

export const FormContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 20px;

  ${media.lessThan('small')`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
  `}
`

export const FormInfo = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    .header {
      h2 {
        font-size: ${theme.font.sizes.xlarge};
      }

      p {
        font-size: ${theme.font.sizes.medium};
        color: ${theme.colors.lightGray};
      }
    }

    .infos {
      a {
        display: flex;
        align-items: center;

        text-decoration: none;
        color: #fff;

        svg {
          margin-right: 10px;
          color: ${theme.colors.primary};
        }

        & + a {
          margin-top: 30px;
        }
      }
    }

    ${media.lessThan('medium')`
      .header {
        margin-bottom: 20px;

        h2 {
          font-size: ${theme.font.sizes.large};
        }

        p {
          font-size: ${theme.font.sizes.small};
        }
      }
    `}
  `}
`

export const Form = styled.form`
  ${({ theme }) => css`
    padding: 20px;
    border-radius: 4px;
    border: 1px solid ${theme.colors.darkGray};
    box-shadow: 0 20px 50px rgba(0, 0, 0, 0.5);

    ${media.lessThan('small')`
      padding: 0;
      border-radius:  0;
      border: 0;
      box-shadow: unset;
    `}
  `}
`

export const Inputs = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 20px;
  margin-bottom: 20px;

  ${media.lessThan('medium')`
    display: flex;
    flex-direction: column;
  `}
`

export const InputContainer = styled.div`
  width: 320px;
  position: relative;

  ${media.lessThan('medium')`
    width: 350px;
  `}
`

export const Label = styled.label`
  ${({ theme }) => css`
    position: absolute;
    left: 10px;
    top: 14px;
    transition: all 0.2s;
    padding: 0 2px;
    z-index: 1;
    color: ${theme.colors.darkBlue};
    opacity: 0;

    &::before {
      content: '';
      height: 5px;
      background: ${theme.colors.darkGray};
      position: absolute;
      left: 0px;
      top: 10px;
      width: 100%;
      z-index: -1;
    }
  `}
`

export const Input = styled.input`
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall};
    width: 100%;
    height: 100%;
    border: 2px solid ${theme.colors.secondary};
    background: ${theme.colors.darkGray};
    border-radius: ${theme.border.radius};
    font-size: ${theme.font.sizes.medium};
    outline: none;
    transition: ${theme.transition.default};
    color: ${theme.colors.white};

    &:focus {
      border: 2px solid ${theme.colors.primary};
    }

    &:focus + .filled {
      top: -10px;
      color: ${theme.colors.primary};
      font-size: ${theme.font.sizes.xsmall};
      opacity: 1;
    }

    &::placeholder {
      font-size: ${theme.font.sizes.small};
      opacity: 1;
      transition: ${theme.transition.default};
    }

    &:focus::placeholder {
      opacity: 1;
    }
  `}
`

export const TextareaContainer = styled.div`
  width: 100%;
  position: relative;
  margin-bottom: 20px;

  ${media.lessThan('medium')`
    width: 350px;
  `}
`

export const TextArea = styled.textarea`
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall};
    width: 100%;
    height: 100%;
    border: 2px solid ${theme.colors.secondary};
    background: ${theme.colors.darkGray};
    border-radius: ${theme.border.radius};
    font-size: ${theme.font.sizes.medium};
    outline: none;
    transition: ${theme.transition.default};
    color: ${theme.colors.white};

    &:focus {
      border: 2px solid ${theme.colors.primary};
    }

    &:focus + .filled {
      top: -10px;
      color: ${theme.colors.primary};
      font-size: ${theme.font.sizes.xsmall};
      opacity: 1;
    }

    &::placeholder {
      font-size: 16px;
      opacity: 1;
      transition: ${theme.transition.default};
    }

    &:focus::placeholder {
      opacity: 1;
    }
  `}
`

export const ButtonForm = styled.button`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;

    width: 100%;
    padding: ${theme.spacings.xxsmall};
    border: 0;
    border-radius: ${theme.border.radius};

    background: ${theme.colors.primary};
    font-size: ${theme.font.sizes.medium};
    transition: ${theme.transition.default};
    cursor: pointer;

    &:hover {
      border: 1px solid ${theme.colors.primary};
      color: ${theme.colors.primary};
      background: transparent;
    }

    svg {
      margin-right: 2px;
    }

    ${media.lessThan('medium')`
      width: 350px;
    `}
  `}
`
