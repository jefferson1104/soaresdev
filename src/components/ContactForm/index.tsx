import { useState } from 'react'

import { SendPlane } from '@styled-icons/remix-fill/SendPlane'
import { Whatsapp as WhatsappIcon } from '@styled-icons/icomoon/Whatsapp'
import { Email as EmailIcon } from '@styled-icons/material-outlined/Email'
import { Location as LocationIcon } from '@styled-icons/entypo/Location'

import * as S from './styles'
import Container from 'components/Container'

const ContactForm = () => {
  const [isActive, setIsActive] = useState(false)
  const [name, setName] = useState('')
  const [lastName, setLastName] = useState('')
  const [email, setEmail] = useState('')
  const [phone, setPhone] = useState('')

  // console.log('NOME =>', name)
  // console.log('SOBRENOME =>', lastName)
  // console.log('EMAIL =>', email)
  // console.log('PHONE =>', phone)

  return (
    <S.Wrapper>
      <h2 data-aos="fade-down">Contato</h2>

      <Container>
        <S.FormContainer data-aos="fade-up">
          <S.FormInfo>
            <div className="header">
              <h2>Informações de contato</h2>
              <p>
                preencha o formulário e nossa equipe entrará em contato com você
                em até 24 horas.
              </p>
            </div>

            <div className="infos">
              <a href="https://wa.me/+5511958002292">
                <WhatsappIcon width={30} /> +55 11 95800-2292
              </a>

              <a href="mailto:jefferson1104junior@gmail.com">
                <EmailIcon width={30} /> jefferson1104junior@gmail.com
              </a>

              <a href="https://goo.gl/maps/cysVrreghC73Tzkh7">
                <LocationIcon width={30} /> São Paulo, SP - Brasil
              </a>
            </div>

            <div />
          </S.FormInfo>

          <S.Form
            onSubmit={() => {
              'testando'
            }}
          >
            <S.Inputs>
              <S.InputContainer>
                <S.Input
                  id="name"
                  type="text"
                  placeholder="Nome"
                  autoComplete="off"
                  value={name}
                  onChange={(event) => setName(event.target.value)}
                  onFocus={() => setIsActive(true)}
                  required
                />
                <S.Label className={isActive ? 'filled' : ''} htmlFor="name">
                  Nome
                </S.Label>
              </S.InputContainer>

              <S.InputContainer>
                <S.Input
                  id="lastname"
                  type="text"
                  placeholder="Sobrenome"
                  autoComplete="off"
                  value={lastName}
                  onChange={(event) => setLastName(event.target.value)}
                  onFocus={() => setIsActive(true)}
                  required
                />
                <S.Label
                  className={isActive ? 'filled' : ''}
                  htmlFor="lastname"
                >
                  Sobrenome
                </S.Label>
              </S.InputContainer>

              <S.InputContainer>
                <S.Input
                  id="email"
                  type="email"
                  placeholder="Email"
                  autoComplete="off"
                  value={email}
                  onChange={(event) => setEmail(event.target.value)}
                  onFocus={() => setIsActive(true)}
                  required
                />
                <S.Label className={isActive ? 'filled' : ''} htmlFor="email">
                  Email
                </S.Label>
              </S.InputContainer>

              <S.InputContainer>
                <S.Input
                  id="phone"
                  type="text"
                  placeholder="Telefone"
                  autoComplete="off"
                  value={phone}
                  onChange={(event) => setPhone(event.target.value)}
                  onFocus={() => setIsActive(true)}
                  required
                />
                <S.Label className={isActive ? 'filled' : ''} htmlFor="phone">
                  Telefone
                </S.Label>
              </S.InputContainer>
            </S.Inputs>

            <S.TextareaContainer>
              <S.TextArea
                id="message"
                placeholder="Mensagem"
                autoComplete="off"
                rows={5}
              />

              <S.Label className={isActive ? 'filled' : ''} htmlFor="message">
                Mensagem
              </S.Label>
            </S.TextareaContainer>

            <S.ButtonForm
              type="submit"
              // disabled={comment.length === 0 || isSendingFeedback}
            >
              {/* {isSendingFeedback ? <Loading /> : 'Send feedback'} */}
              <SendPlane size={20} />
              Enviar
            </S.ButtonForm>
          </S.Form>
        </S.FormContainer>
      </Container>
    </S.Wrapper>
  )
}

export default ContactForm
