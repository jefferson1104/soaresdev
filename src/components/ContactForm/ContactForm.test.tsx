import { render, screen } from '@testing-library/react'

import ContactForm from '.'

describe('ContactForm component', () => {
  it('should render the heading', () => {
    const { container } = render(<ContactForm />)

    expect(
      screen.getByRole('heading', { name: /ContactForm/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
