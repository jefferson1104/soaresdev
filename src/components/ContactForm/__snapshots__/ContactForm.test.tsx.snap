// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component should render the heading 1`] = `
<main
  class=""
>
  <h1>
    ContactForm
  </h1>
</main>
`;
